--- output/build/linux-5.11/arch/arm/boot/dts/sun8i-v3s.dtsi	2021-02-14 22:32:24.000000000 +0000
+++ linux-5.11/arch/arm/boot/dts/sun8i-v3s.dtsi	2022-09-16 17:26:43.599479851 +0100
@@ -161,6 +161,20 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges;
+
+			sram_c: sram@1d00000 {
+				compatible = "mmio-sram";
+				reg = <0x01d00000 0x80000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0 0x01d00000 0x80000>;
+
+				ve_sram: sram-section@0 {
+					compatible = "allwinner,sun8i-v3s-sram-c1",
+						     "allwinner,sun4i-a10-sram-c1";
+					reg = <0x000000 0x80000>;
+				};
+			};
 		};
 
 		tcon0: lcd-controller@1c0c000 {
@@ -197,6 +211,16 @@
 			};
 		};
 
+        video-codec@1c0e000 {
+			compatible = "allwinner,sun8i-v3s-video-engine";
+			reg = <0x01c0e000 0x1000>;
+			clocks = <&ccu CLK_BUS_VE>, <&ccu CLK_VE>,
+				 <&ccu CLK_DRAM_VE>;
+			clock-names = "ahb", "mod", "ram";
+			resets = <&ccu RST_BUS_VE>;
+			interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+			allwinner,sram = <&ve_sram 1>;
+		};
 
 		mmc0: mmc@1c0f000 {
 			compatible = "allwinner,sun7i-a20-mmc";
@@ -369,6 +393,25 @@
 				function = "uart2";
 			};
 
+            lcd_pd_pins: lcd-pd-pins {
+				pins = "PD0", "PD1", "PD2", "PD3", "PD4", "PD5", /* R */
+				       "PD6", "PD7", "PD8", "PD9", "PD10", "PD11", /* G */
+				       "PD12", "PD13", "PD14", "PD15", "PD16", "PD17", /* B */
+				       "PD18", /* CLK */
+                       "PD19", /* DE */
+                       "PD20", /* HSYNC */
+                       "PD21"; /* VSYNC */
+				function = "lcd";
+			};
+
+            lcd_pe_pins: lcd-pe-pins {
+				pins = "PE0", "PE1", "PE2", "PE3", "PE4", "PE5",
+				       "PE6", "PE7", "PE8", "PE9", "PE10", "PE11",
+				       "PE12", "PE13", "PE14", "PE15", "PE16", "PE17",
+				       "PE18", "PE19", "PE23", "PE24";
+				function = "lcd";
+			};
+
 			mmc0_pins: mmc0-pins {
 				pins = "PF0", "PF1", "PF2", "PF3",
 				       "PF4", "PF5";
@@ -385,6 +428,16 @@
 				bias-pull-up;
 			};
 
+            pwm0_pins: pwm0-pins {
+				pins = "PB4";
+				function = "pwm0";
+			};
+
+			pwm1_pins: pwm1-pins {
+				pins = "PB5";
+				function = "pwm1";
+			};
+
 			spi0_pins: spi0-pins {
 				pins = "PC0", "PC1", "PC2", "PC3";
 				function = "spi0";
@@ -400,6 +453,14 @@
 			clocks = <&osc24M>;
 		};
 
+        pwm: pwm@1c21400 {
+			compatible = "allwinner,sun7i-a20-pwm";
+			reg = <0x01c21400 0x400>;
+			clocks = <&osc24M>;
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
 		wdt0: watchdog@1c20ca0 {
 			compatible = "allwinner,sun6i-a31-wdt";
 			reg = <0x01c20ca0 0x20>;
