diff --color -ruN linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi output/build/linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi
--- linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi	2022-10-20 17:47:36.559050924 +0100
+++ output/build/linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi	2022-10-20 17:52:12.946678035 +0100
@@ -476,6 +476,16 @@
 				bias-pull-up;
 			};
 
+			pwm0_pins: pwm0-pins {
+				pins = "PB4";
+				function = "pwm0";
+			};
+
+			pwm1_pins: pwm1-pins {
+				pins = "PB5";
+				function = "pwm1";
+			};
+
 			spi0_pins: spi0-pins {
 				pins = "PC0", "PC1", "PC2", "PC3";
 				function = "spi0";
