diff --color -ruN linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi output/build/linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi
--- linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi	2022-10-20 17:14:37.517810231 +0100
+++ output/build/linux-6.0.1/arch/arm/boot/dts/sun8i-v3s.dtsi	2022-10-20 17:15:07.141692722 +0100
@@ -163,6 +163,20 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges;
+
+			sram_c: sram@1d00000 {
+				compatible = "mmio-sram";
+				reg = <0x01d00000 0x80000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0 0x01d00000 0x80000>;
+
+				ve_sram: sram-section@0 {
+					compatible = "allwinner,sun8i-v3s-sram-c1",
+					"allwinner,sun4i-a10-sram-c1";
+					reg = <0x000000 0x80000>;
+				};
+			};
 		};

 		nmi_intc: interrupt-controller@1c000d0 {
@@ -217,6 +231,16 @@
 			};
 		};

+		video-codec@1c0e000 {
+			compatible = "allwinner,sun8i-v3s-video-engine";
+			reg = <0x01c0e000 0x1000>;
+			clocks = <&ccu CLK_BUS_VE>, <&ccu CLK_VE>,
+			<&ccu CLK_DRAM_VE>;
+			clock-names = "ahb", "mod", "ram";
+			resets = <&ccu RST_BUS_VE>;
+			interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+			allwinner,sram = <&ve_sram 1>;
+		};

 		mmc0: mmc@1c0f000 {
 			compatible = "allwinner,sun7i-a20-mmc";
