--- output/build/linux-5.11/arch/arm/boot/dts/sun8i-s3-elimo-initium.dts	2021-02-14 22:32:24.000000000 +0000
+++ linux-5.11/arch/arm/boot/dts/sun8i-s3-elimo-initium.dts	2022-09-16 16:59:50.259376720 +0100
@@ -14,6 +14,26 @@
 	aliases {
 		serial1 = &uart1;
 	};
+
+    backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm 0 1000000 0>;
+		brightness-levels = <0 30 40 50 60 70 100>;
+		default-brightness-level = <2>;
+        enable-gpios = <&pio 4 18 GPIO_ACTIVE_HIGH>; /* PE18 */
+	};
+
+	panel: panel {
+		backlight = <&backlight>;
+
+		port {
+			panel_input: endpoint {
+				remote-endpoint = <&tcon0_out_lcd>;
+			};
+		};
+	};
+
+
 };
 
 &uart1 {
@@ -27,3 +47,50 @@
 	phy-mode = "mii";
 	status = "okay";
 };
+
+/* i2c bus for the touchscreen */
+&i2c0 {
+	status = "okay";
+
+    /* touchscreen controller on the NTW panel */
+    edt-ft5x06@38 {
+        compatible = "edt,edt-ft5406";
+        reg = <0x38>;
+        interrupt-parent = <&pio>;
+        interrupts = <1 13 IRQ_TYPE_EDGE_FALLING>; /* PB13 */
+        reset-gpios = <&pio 4 20 GPIO_ACTIVE_LOW>; /* PE20 */
+    };
+};
+
+/* enable the display engine */
+&de {
+	status = "okay";
+};
+
+/* PWM controller for the backlight */
+&pwm {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pwm0_pins>; /* PB4 */
+	status = "okay";
+};
+
+/* LCD panel (the compatible string depends on a patched kernel,
+ * they are not available in the mainline kernel)
+ */
+&panel {
+	compatible = "ntw,500t40d007a";
+};
+
+/* panel controller (LVDS,parallel RGB, serial RGB...) */
+&tcon0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&lcd_pd_pins>;
+	status = "okay";
+};
+
+&tcon0_out {
+	tcon0_out_lcd: endpoint@0 {
+		reg = <0>;
+		remote-endpoint = <&panel_input>;
+	};
+};
